client/modules/wj-util/src/index.ts

```diff
@@ -430,3 +434,42 @@ export function html(strings: TemplateStringsArray, ...subs: (string | string[])
 export function mod(a: number, n: number) {
   return ((a % n) + n) % n
 }
+
+/**
+ * Replaces a range inside of a string with a substitute.
+ *
+ * @param str - The string which should have a range inside of it replaced.
+ * @param from - The start of the replacement range.
+ * @param to - The end of the replacement range.
+ * @param sub - The replacement/substitute string.
+ */
+export function replaceRange(str: string, from: number, to: number, sub: string) {
+  return str.substr(0, from) + sub + str.substr(to)
+}
+
+/**
+ * Capitalizes (or lowercases, if set) a string. Does not uppercase the
+ * entire string, just the first character.
+ *
+ * @param str - The string to capitalize.
+ * @param state - Whether to capitalize or lowercase. Defaults to capitalize.
+ */
```

Comment:
All of these only work on ASCII and should be noted as such.

If we want to check case-insensitively, you will need to do a unicode normalize/case fold.

Reply:
I thought `toLowerCase` and `toUpperCase` accounted for the characters it was checking, but if it doesn't I think JS has another function for that.

one google search later:

Okay, so `toLowerCase` and `toUpperCase` do correctly handle non-ASCII text, however certain locales handle the same text in different ways, such as the `TR` locale. For that, there is `toLocaleLowerCase` and `toLocaleUpperCase`, which about 99% of the time is going to give the same result as the ordinary functions. I could hook that up into these functions, using an optional `locale` argument, I think.

Reply:
Okay, cool. Can you leave this as a comment here? Since it looks like a naive ASCII implementation at first glance.

Reply:
donezo, added with locale support